### PR TITLE
test: added more e2e permissionless tests

### PR DESCRIPTION
**Pull request by bermuell**
## Description

Add more e2e tests for permissionless
- removal of permissionless chains
-  transformation of TopN chain to permissionless chain
- add permissionless to nightly e2e tests

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Provided a link to the relevant issue or specification
- [x] Reviewed "Files changed" and left comments if necessary 
- [ ] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] Confirmed all author checklist items have been addressed
- [ ] Confirmed that this PR does not change production code

### CODE REVIEW

File: tests/e2e/actions.go [hunk 1]
```diff
@@ -1981,12 +2035,11 @@ func (tr Chain) relayPacketsHermes(
...
 ) {
-	// Because `.app_state.provider.params.blocks_per_epoch` is set to 3 in the E2E tests, we wait 4 blocks
+	// Because `.app_state.provider.params.blocks_per_epoch` is set to 3 in the E2E tests, we wait 3 blocks
 	// before relaying the packets to guarantee that at least one epoch passes and hence any `VSCPacket`s get
 	// queued and are subsequently relayed.
-	tr.waitBlocks(action.ChainA, 4, 90*time.Second)
-	tr.waitBlocks(action.ChainB, 4, 90*time.Second)
-
+	tr.waitBlocks(action.ChainA, 3, 90*time.Second)
+	tr.waitBlocks(action.ChainB, 3, 90*time.Second)
```
Comment (bermuell) [inline tests/e2e/actions.go:2042]: going back to original #blocks to wait for.

**Comment by sainoe** [comment]
LGTM!

**Review by sainoe — APPROVED** [review]

File: tests/e2e/steps_permissionless_ics.go [hunk 2]
```diff
@@ -52,7 +53,11 @@ func stepsPermissionlessICS() []Step {
 						TopN: 0,
 					},
 				},
-				State: State{},
+				State: State{
+					ChainID("provi"): e2e.ChainState{
+						ConsumerChains: &map[ChainID]bool{"cons2": true}, // Consumer chain "consu1" is now removed
```
Comment (insumity) [inline tests/e2e/steps_permissionless_ics.go:58]: Why does the comment state that it "is now removed"? It seems that it was just added.

File: tests/e2e/steps_permissionless_ics.go [hunk 3]
```diff
@@ -157,8 +166,28 @@ func stepsPermissionlessICS() []Step {
 					},
 					ChainID("provi"): e2e.ChainState{
 						HasToValidate: &map[ValidatorID][]ChainID{
-							ValidatorID("alice"): {"consu"},
-							ValidatorID("bob"):   {"consu"}, // bob is still a validator on consu chain
+							ValidatorID("alice"): {"cons1"},
+							ValidatorID("bob"):   {"cons1"}, // bob is still a validator on consu chain
```
Comment (insumity) [inline tests/e2e/steps_permissionless_ics.go:170]:
```suggestion
							ValidatorID("bob"):   {"cons1"}, // bob is still a validator on cons1 chain
```
?

File: tests/e2e/steps_permissionless_ics.go [hunk 4]
```diff
@@ -157,8 +166,28 @@ func stepsPermissionlessICS() []Step {
...
+		[]Step{
+			{
+				Action: RemoveConsumerChainAction{
+					Chain:         ChainID("provi"),
+					From:          ValidatorID("alice"),
+					ConsumerChain: ChainID("cons1"),
+				},
+				State: State{
+					ChainID("provi"): e2e.ChainState{
+						ConsumerChains: &map[ChainID]bool{"cons2": true}, // Consumer chain "consu1" is now removed
```
Comment (insumity) [inline tests/e2e/steps_permissionless_ics.go:187]:
```suggestion
						ConsumerChains: &map[ChainID]bool{"cons2": true}, // Consumer chain "cons1" is now removed
```

File: tests/e2e/steps_permissionless_ics.go [hunk 5]
```diff
@@ -157,8 +166,28 @@ func stepsPermissionlessICS() []Step {
...
+						HasToValidate: &map[ValidatorID][]ChainID{
+							ValidatorID("alice"): {"cons1"},
+							ValidatorID("bob"):   {"cons1"}, // bob is still a validator on consu chain
```
Comment (insumity) [inline tests/e2e/steps_permissionless_ics.go:190]:
```suggestion
							ValidatorID("bob"):   {"cons1"}, // bob is still a validator on cons1 chain
```
?

Comment (insumity) [reply tests/e2e/steps_permissionless_ics.go:190]: Actually, `cons1` is removed but validators still have to validate according to this query. This seems wrong: Why would you need to validate a chain that has been stopped? I'll publish a fix on this in a bit.

**Review by insumity — APPROVED** [review]
Nice!

LGTM

File: tests/e2e/steps_permissionless_ics.go (hunk 5 above)
Comment (bermuell) [reply tests/e2e/steps_permissionless_ics.go:190]: this is based on the description from the ADR that the final delete message will be called after unbonding period has passed.

**Stopped phase.** 
Lastly, the owner of a chain can choose to exit ICS by executing a `MsgRemoveConsumer`.
After some period of time (e.g., provider's unbonding period), all state related to the stopped consumer chain can be removed.

As discussed , we can open an issue and change the behaviour to what HasToValidate() should return, e.g. only report validators of active chains.

File: tests/e2e/steps_permissionless_ics.go (hunk 2 above)
Comment (bermuell) [reply tests/e2e/steps_permissionless_ics.go:58]: wrong line, it was not even started yet.

**Pull request merged**